### ISSUE TITLE
allow region-based functionality to use edge padding.

### ISSUE THREAD

**Issue by AndrewIngram** [airbnb/react-native-maps#1174]
There are scenarios where part of the map is obscured (either off-screen, or by another component). I've managed to find an old question on stackoverflow that illustrates a similar use case to mine but on web:

http://stackoverflow.com/questions/10656743/how-to-offset-the-center-point-in-google-maps-api-v3

This kind of result can be achieved with react-native-maps if using `fitToCoordinates`, because it accepts an `edgePadding` option. But i'm struggling to see a simple way to get the same effect when using `animateToRegion` `region` or `initialRegion`.

I'm not sure if it makes sense as an API, but one idea would be to extend the region type to include an `edgePadding` field.

**Comment by lsps9150414** [comment]
Agree. IMHO all `fitToCoordinates`, `fitToSuppliedMarkers`, `fitToElements` should take the same `options: { edgePadding: EdgePadding, animated: Boolean }` param.

**Comment by peterjuras** [comment]
I've opened a new PR for to add setPadding support via the "mapPadding" prop:

https://github.com/airbnb/react-native-maps/pull/1750
